design/security-model.md

```diff
@@ -0,0 +1,360 @@
+  resources: ["clusterroles", "clusterrolebindings", "roles", "rolebindings"]
+  verbs: [ "get", "list", "watch", "create", "update", "delete" ]
+- apiGroups: ["storage.k8s.io"]
+  resources: ["storageclasses"]
+  verbs: [ "get", "list", "watch", "delete" ]
+- apiGroups: ["rook.io"]
+  resources: ["*"]
+  verbs: [ "*" ]
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: rook-operator
+  namespace: rook-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: rook-operator
+subjects:
+- kind: ServiceAccount
+  name: rook-operator
+  namespace: rook-system
+```
+
+`rook-operator` is a highly privileged service account with cluster wide scope. It likely has more privileges than is currently needed, for example, the operator does not create namespaces today. Note the name `rook-system` and `rook-operator` are not important and can be set to anything.
+
+Once the rook operator is up and running it will automatically create the service account for the rook agent and the following RBAC rules:
+
+```
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: rook-agent
+  namespace: rook-system
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: rook-agent
+rules:
+- apiGroups: [""]
+  resources: ["pods", "secrets", "configmaps", "persistenvolumes", "nodes", "nodes/proxy"]
+  verbs: [ "get", "list" ]
+- apiGroups: ["storage.k8s.io"]
+  resources: ["storageclasses"]
+  verbs: [ "get" ]
+- apiGroups: ["rook.io"]
+  resources: ["volumeattachment"]
+  verbs: [ "get", "list", "watch", "create", "update" ]
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: rook-agent
+  namespace: rook-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: rook-agent
+subjects:
+- kind: ServiceAccount
+  name: rook-agent
+  namespace: rook-system
+```
+
+When the cluster admin create a new Rook cluster they do so by adding a namespace and the rook cluster spec:
+
+```
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: mycluster
+---
+apiVersion: rook.io/v1alpha1
+kind: Cluster
+metadata:
+  name: myrookcluster
+  namespace: mycluster
+	...
+```
+
+At this point the rook operator will notice that a new rook cluster CRD showed up and proceeds to create a service account for the `rook-api` and `rook-ceph-osd`. It will also use the `default` service account in the `mycluster` namespace for some pods.
+
+The `rook-api` service account and RBAC rules are as follows:
+
+```
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: rook-api
+  namespace: mycluster
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: rook-api
+  namespace: mycluster
+rules:
+- apiGroups: [""]
+  resources: ["namespaces", "secrets", "pods", "services", "nodes", "configmaps", "events"]
+  verbs: [ "get", "list", "watch", "create", "update" ]
```

Comment:
Why nodes?

Reply:
I dont think `nodes` is needed. In fact this whole `rook-api` role should go away.